### PR TITLE
fix(relay): Increase rate-limit on auth endpoints [INGEST-1310]

### PR DESCRIPTION
**Pull request by untitaker**
Internal Relays require the authentication endpoints during deployments.
We need to make sure those endpoints are always available within reason.

At the same time, external Relays operated by customers will also access
the same endpoints repeatedly. To avoid overloading internal systems, we
keep a higher rate limit of 200/s in place.

### CODE REVIEW

**Review by AniketDas-Tekky — APPROVED** [review]

**Review by jan-auer — CHANGES_REQUESTED** [review]
We want those auth endpoints rate limited, since they are also being used by external Relays. In a proper fix, we would need to ensure that those rate limits are bypassed by trusted relays, but enforced on external ones. 

See [INGEST-1310](https://getsentry.atlassian.net/browse/INGEST-1310)

**Comment by untitaker** [comment]
I raised the rate limit to 200 rps / IP. I suspect our internal relays's requests (all pods) are coming all from the same IP, so this would be effectively a global limit.

I believe the original limit was 70/IP in code, but our internal stats don't show this kind of traffic.

File: src/sentry/api/endpoints/relay/register_challenge.py [hunk 1]
```diff
@@ -22,6 +23,15 @@ class RelayRegisterChallengeEndpoint(Endpoint):
     authentication_classes = ()
     permission_classes = ()
 
+    enforce_rate_limit = True
+    rate_limits = {
+        "default": {
+            RateLimitCategory.IP: RateLimit(200, 1),
```
Comment (jan-auer) [inline src/sentry/api/endpoints/relay/register_challenge.py:29]: _nit_: Add a constant somewhere in `endpoints/relay/` to indicate which limits belong together.

**Review by jan-auer — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]
Might want to update the PR title / description to something like "Raise rate limits.."

**Pull request merged**